src/client/testing/page-objects/view-availability-appointment-pages/month-view-availability-page.ts

```diff
@@ -1,17 +1,122 @@
+    });
   }
 
-  async verifyViewMonthDisplayed() {
+  async verifyViewNextMonthButtonDisplayed() {
     await expect(this.nextButton).toBeEnabled();
   }
+
+  async verifyAllWeekCardInformationDisplayedCorrectly(
+    expectedWeekOverviews: weekOverview[],
+  ) {
+    for (let i = 0; i < expectedWeekOverviews.length; i++) {
+      const weekOverview = expectedWeekOverviews[i];
+
+      const cardDiv = this.page
+        .getByRole('heading', {
+          name: weekOverview.header,
+        })
+        .locator('..');
+
+      const header = cardDiv.getByRole('heading', {
+        name: weekOverview.header,
+      });
+
+      //assert header
+      await expect(header).toBeVisible();
+
+      if (weekOverview.services.length > 0) {
+        //assert no availability not visible
+        expect(cardDiv.getByText('No availability')).not.toBeVisible();
+
+        //only do for a single service for now!!
+        const serviceCell = cardDiv.getByRole('cell', {
+          name: weekOverview.services[0].serviceName,
+        });
+        const bookedAppointmentsCell = cardDiv.getByRole('cell', {
+          name: weekOverview.services[0].bookedAppointments.toString(),
+        });
+
+        await expect(serviceCell).toBeVisible();
+        await expect(bookedAppointmentsCell).toBeVisible();
+
+        //totals
+        await expect(
+          cardDiv.getByText(
+            `Total appointments: ${weekOverview.totalAppointments}`,
+          ),
+        ).toBeVisible();
+        await expect(
+          cardDiv.getByText(`Booked: ${weekOverview.booked}`, {
+            exact: true,
+          }),
+        ).toBeVisible();
+        await expect(
+          cardDiv.getByText(`Unbooked: ${weekOverview.unbooked}`, {
+            exact: true,
+          }),
+        ).toBeVisible();
+      } else {
+        //no services = no availability
+        await expect(cardDiv.getByText('No availability')).toBeVisible();
+        //totals
+        await expect(cardDiv.getByText('Total appointments: 0')).toBeVisible();
+        await expect(
+          cardDiv.getByText('Booked: 0', { exact: true }),
+        ).toBeVisible();
+        await expect(
+          cardDiv.getByText('Unbooked: 0', { exact: true }),
+        ).toBeVisible();
+      }
+    }
+  }
+
+  async verifyViewMonthDisplayed(requiredWeek: string) {
+    await this.verifyViewNextMonthButtonDisplayed();
+    await expect(
+      this.page
+        .getByRole('listitem')
+        .filter({ has: this.page.getByText(`${requiredWeek}`) }),
+    ).toBeVisible();
+  }
+
+  async openWeekViewHavingDate(requiredWeek: string) {
+    await this.page
+      .getByRole('listitem')
+      .filter({ has: this.page.getByText(`${requiredWeek}`) })
+      .getByRole('link', { name: 'View week' })
+      .click();
+  }
+
+  async navigateToRequiredMonth(month: string) {
+    await this.page.goto(
+      `/manage-your-appointments/site/6877d86e-c2df-4def-8508-e1eccf0ea6be/view-availability?date=${month}`,
```

Comment:
Could we parameterise this id? 
Again, in the future I'm envisaging page objects having a Site property that gets populated on transition but at least for now I think this should be supplied by the test itself